src-v2/NTS.Domain.Core/Objects/Start.cs

```diff
@@ -10,39 +12,34 @@ public Start(
         int loopNumber,
         double distance,
         double totalDistance,
-        Timestamp startAt
+        TimeSpan startAt
     )
     {
         Athlete = athlete;
         Number = number;
         PhaseNumber = loopNumber;
         Distance = distance;
-        TotalDistance = totalDistance.FloorWholeNumberToTens();
+        TotalDistance = totalDistance.RoundWholeNumberToTens();
         StartAt = startAt;
     }
 
+    public Start(Participation participation)
+    {
+        Athlete = participation.Combination.Name;
+        Number = participation.Combination.Number;
+        var phaseIndex = participation.Phases.IndexOf(participation.Phases.Current);
+        PhaseNumber = phaseIndex + 2;
+        Distance = participation.Phases[phaseIndex].Length;
+        TotalDistance = participation.Phases.Distance;
+        StartAt = participation.Phases[phaseIndex + 1].StartTime!.DateTime.TimeOfDay;
+    }
+
     public Person Athlete { get; private set; }
     public int Number { get; private set; }
     public int PhaseNumber { get; private set; }
     public double Distance { get; private set; }
     public double TotalDistance { get; private set; }
-    public Timestamp? StartAt { get; private set; }
-
-    public string StartIn()
-    {
-        if (StartAt == null)
-        {
-            return string.Empty;
-        }
-        if (StartAt > Timestamp.Now())
-        {
-            return (StartAt - Timestamp.Now())!.ToString();
-        }
-        else
-        {
-            return "-" + (StartAt - Timestamp.Now())!.ToString();
-        }
-    }
+    public TimeSpan StartAt { get; private set; }
```

Comment:
Describes a point in time, rather than an interval of time.

Reply:
I can see you're still having trouble with naming and this is a good example: the object which you are working in is named `Start`, which means that simply naming this `Time` is good enough as you get the context of `Start.Time` which describes the information stored accurately. Opposingly, `History` and `Upcoming` aren't good component names, because when navigating the entire solution one is often searching by file name/class name neither `Upcoming` nor `History` provides enough information on their own, so `StartlistUpcoming` and `StartlistHistory` would be better.